Update Kiteworks migration documentation (#1382)
owncloud/docs-server: modules/admin_manual/pages/maintenance/migrating_to_kiteworks.adoc

The following limitations impact the migration process:

* The ownCloud migration app is only functional on `amd64` platforms.

* While ownCloud fully respects letter casing for file and folder names, Kiteworks does not distinguish casing. If case conflicts happen during the migration process, a migration log file describing rclone responses and casing conflicts for files or directories is created. The ownCloud admin must resolve the conflicts to finalize the migration. For details see the xref:migration[Migration] description.

* When group shares have been defined, groups will not get created in Kiteworks. Instead, each member of the group will get an individual user share to the object shared.

* In ownCloud, users can login either using their display name, login name or email address. Kiteworks only allows login using the email address. The presence of the users email address in ownCloud, which must be unique, is therefore a mandatory requirement.

> * Expiry dates for shares are not migrated.

Reviewer: This statement can be miss understood.

Which applies to public links only which are not supported in kiteworks. 
Expiry dates on shared resource in kiteworks is possible but not supported by owncloud.

Author: fixed

=== Kiteworks

> * As a major prerequisite, the Kiteworks instance *must* be running on the "Venice" release or higher.

Reviewer: Latest changes require `Zion`

Author: fixed

* Secret key +
`KW_SECRET`

> NOTE: We recommend having the values saved as shell variables as proposed for ease of use in the following commands. 

Reviewer: the env vars above? Pointless - they are only used once - why save them in an env var?

Author: fixed

=== Migration

After all prerequisites, installations, configurations and the verification has passed, you can initiate the migration process. The migration is split into four parts which are:

[,subs="+callouts,macros,attributes+"]
----
1. Migrate users <1>
2. Migrate files <1>
3. Migrate shares <2>
4. Disable users <3>
----

> <1> These steps migrate all ordinary, guest and disabled users and files. You can rerun these steps to migrate any items that were not part when the respective step was called. Note that the steps must be made in that order.

Reviewer 2: ```suggestion
<1> These steps migrate all ordinary, guest and disabled users and files. You can rerun these steps to migrate any items that did not exist when the respective step was called. Note that the steps must be made in that order.
```

Or maybe "that did not exist, or failed to migrate when..."

Author: I have updated you suggestion to `that did not exist, or failed to migrate` so I can commit it.

<2> This step is a breaking change and migrates all shares. When this step has run, the former steps cant be run again.
<3> This step finalizes the migration by disabling all users on Kiteworks that are disabled on ownCloud.

> The migration step transferring files will naturally take its time depending on the amount of data and bandwidth available. All other steps will complete relative quickly as only a few or metadata is transferred.

Reviewer 2: ```suggestion
The migration step transferring files will naturally take its time depending on the amount of data and bandwidth available. All other steps will complete relative quickly as only metadata is transferred.
```

NOTE: All migration steps, especially files, can be interrupted harmlessly at any time. Starting a new migration run will continue where the previous one stopped.

==== Migrate Users

Issue the following command to start migrating users:

[source,bash]
----
sudo -u www-data \
  php /var/www/owncloud/occ \
  migrate:to-kiteworks:users \
  $KW_ADMIN_USER \
  $KW_PROFILE_GUEST (optional)
----

Note that you can optionally add a Kiteworks guest user profile that will be assigned to guest users defined in owncloud when migrating.

The command does not require user interaction. It can be run e.g. as a screen session so that reported issues can be seen directly or as a background job.

==== Migrate Files

Issue the following command to start migrating files:

[source,bash]
----
sudo -u www-data \
  php /var/www/owncloud/occ \
  migrate:to-kiteworks:files \
  $KW_ADMIN_USER
----

For a possible improvement of transfer performance when migrating files, read the xref:tuning-transfer-performance[Tuning Transfer Performance] section below.

> The command does not require user interaction. It can be run e.g. as a screen session so that reported issues can be seen directly or as a background job. During the migration process, a log file named `migrate-kiteworks-<timestamp>.csv` is created in the ownCloud root folder. This file contains:

Reviewer: migrate-kiteworks-files<timestamp>.csv

Author: fixed

As you can see above, there is `Duplicate` notice for a file and another one for a directory name. `Duplicate` notices are logged for case conflicts. A conflict takes place because a file or directory that has been migrated earlier is in conflict with the name of the reported object. The conflicts for the particular users need to be resolved within ownCloud. When this is done, the migration can be restarted. `rclone` will compare both sides to identify already migrated objects and will continue with those objects that have not been migrated yet.

==== Migrate Shares

IMPORTANT: When running this command, you cant run the `migrate:files` command anymore!

Issue the following command to start migrating shares:

[source,bash]
----
sudo -u www-data \
  php /var/www/owncloud/occ \
  migrate:to-kiteworks:shares \
  $KW_ADMIN_USER
----

The command does not require user interaction. It can be run e.g. as a screen session so that reported issues can be seen directly or as a background job.

==== Disable Users

> When there are disabled users on the ownCloud side, these users will be migrated as ordinary users to Kitworks but are enabled. This is due the fact that a disabled user can provide shares. As final step, run this command to disable users at Kitworks that are diabled at ownCloud.

Reviewer 2: ```suggestion
When there are disabled users on the ownCloud side, these users will be migrated as ordinary users to Kiteworks but are enabled. This is due the fact that a disabled user can provide shares. As a final step, run this command to disable users at Kiteworks that are disabled at ownCloud.
```

As you can see above, there is `Duplicate` notice for a file and another one for a directory name. `Duplicate` notices are logged for case conflicts. A conflict takes place because a file or directory that has been migrated earlier is in conflict with the name of the reported object. The conflicts for the particular users need to be resolved within ownCloud. When this is done, the migration can be restarted. `rclone` will compare both sides to identify already migrated objects and will continue with those objects that have not been migrated yet.

==== Migrate Shares

IMPORTANT: When running this command, you cant run the `migrate:files|users` command anymore!

Issue the following command to start migrating shares:

[source,bash]
----
sudo -u www-data \
  php /var/www/owncloud/occ \
  migrate:to-kiteworks:shares \
  $KW_ADMIN_USER
----

> The command does not require user interaction. It can be run e.g. as a screen session so that reported issues can be seen directly or as a background job.

Reviewer: this command creates a migrate-kiteworks-shares-<timestamp>.csv

Author: fixed

Performance can be tuned with the environment variable `RCLONE_TRANSFERS` which defines the number of concurrent file uploads.

> The following example command is using 16 parallel transfers:

Reviewer: drop this please from the official docs - we only tested with default settings.

Author: fixed
